### ISSUE TITLE
bug[valid-expect]: false positive when using the "branded" accessor

### ISSUE THREAD

**Issue by eranhirsch** [vitest-dev/eslint-plugin-vitest#539]
The rule is surfacing errors for lines like:

```ts
expectTypeOf(result).branded.toEqualTypeOf<MyType>();
```

With the error

> "Expect has unknown modifier".

The `branded` accessor is not documented in the vitest docs, but is supported by the library, and is part of expect-type:

https://github.com/mmkal/expect-type/blob/main/README.md?plain=1#L608-L655
